### PR TITLE
[#PAB-362] add download endpoint

### PR DESCRIPTION
**Pull request by cyrusdobbs**
https://trello.com/c/wBlZvwpn/362-create-data-as-file-api-endpoint

### Change description
Adds an endpoint that returns some data in either xlsx or json format. The data returned is just example data for now.

- [X] Unit tests and other appropriate tests added or updated
- [X] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### CODE REVIEW

File: openapi/api.yml [hunk 1]
```diff
@@ -6,20 +6,31 @@ info:
...
 paths:
-  /data:
+  /download:
     get:
-      operationId: core.controllers.examples.example
-      summary: Returns a list of data.
-      description: Optional extended description in CommonMark or HTML.
+      operationId: core.controllers.download.download
+      summary: Returns some data in a binary format.
+      description: Takes a file format and returns some data in that format.
```
Comment (gidsg) [inline openapi/api.yml:13]: Guess it's only binary if excel, so maybe say, 'in requested format' or similar?

**Review by nsnape — COMMENTED** [review]
LGTM, nice and simple. I guess this will hopefully just work when we want to output the updated DM data? 

The only thing we may need to add is a little transformation before it gets spat out, but handy if it's in a Dict of dataframes

File: openapi/api.yml (hunk 1 above)
Comment (cyrusdobbs) [reply openapi/api.yml:13]: changed this

**Review by gidsg — APPROVED** [review]
LGTM 👍 🌸

**Comment by cyrusdobbs** [comment]
> LGTM, nice and simple. I guess this will hopefully just work when we want to output the updated DM data?
> The only thing we may need to add is a little transformation before it gets spat out, but handy if it's in a Dict of dataframes

It will work seamlessly for excel but for json we'll need to change the call to the Package model to the Programme model, but yeah. It will need more changes than this though obvs because at the moment it just grabs _something_ from the db and returns it because we dont care about the contents yet

**Pull request merged**